src/client/wallet/PowerUpOrDown.js

```diff
@@ -0,0 +1,213 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { injectIntl, FormattedMessage } from 'react-intl';
+import { Form, Input, Modal } from 'antd';
+import SteemConnect from '../steemConnectAPI';
+import { closePowerUpOrDown } from './walletActions';
+import {
+  getAuthenticatedUser,
+  getIsPowerUpOrDownVisible,
+  getIsPowerDown,
+  getTotalVestingShares,
+  getTotalVestingFundSteem,
+} from '../reducers';
+import formatter from '../helpers/steemitFormatter';
+import './Transfer.less';
+
+@injectIntl
+@connect(
+  state => ({
+    visible: getIsPowerUpOrDownVisible(state),
+    user: getAuthenticatedUser(state),
+    totalVestingShares: getTotalVestingShares(state),
+    totalVestingFundSteem: getTotalVestingFundSteem(state),
+    down: getIsPowerDown(state),
+  }),
+  {
+    closePowerUpOrDown,
+  },
+)
+@Form.create()
+export default class PowerUpOrDown extends React.Component {
+  static propTypes = {
+    intl: PropTypes.shape().isRequired,
+    form: PropTypes.shape().isRequired,
+    visible: PropTypes.bool.isRequired,
+    closePowerUpOrDown: PropTypes.func.isRequired,
+    user: PropTypes.shape().isRequired,
+    totalVestingShares: PropTypes.string.isRequired,
+    totalVestingFundSteem: PropTypes.string.isRequired,
+    down: PropTypes.bool.isRequired,
+  };
+
+  static amountRegex = /^[0-9]*\.?[0-9]{0,6}$/;
+
+  state = {
+    oldAmount: undefined,
+  };
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.down !== this.props.down) {
+      this.props.form.setFieldsValue({
+        amount: '',
+      });
+    }
+  }
+
+  getAvailableBalance = () => {
+    const { user, down, totalVestingShares, totalVestingFundSteem } = this.props;
+
+    return down
+      ? formatter.vestToSteem(user.vesting_shares, totalVestingShares, totalVestingFundSteem)
+      : parseFloat(user.balance);
+  };
+
+  handleBalanceClick = event => {
+    const { oldAmount } = this.state;
+    const value = parseFloat(event.currentTarget.innerText);
+    this.setState({
+      oldAmount: PowerUpOrDown.amountRegex.test(value) ? value : oldAmount,
+    });
+    this.props.form.setFieldsValue({
+      amount: value,
+    });
+  };
+
+  handleContinueClick = () => {
+    const { form, user, down, totalVestingShares, totalVestingFundSteem } = this.props;
+    form.validateFields({ force: true }, (errors, values) => {
+      const vests = (
+        values.amount / formatter.vestToSteem(1, totalVestingShares, totalVestingFundSteem)
+      ).toFixed(6);
```

Comment:
we're fixing vests to 6 decimal places?

Reply:
Otherwise transaction would be invalid:
`decimals == 6: Incorrect decimal places`